second commit

LikeARivers opened the pull request:
Добавлены tests, locators, settings, data

LikeARivers committed:
second commit

LikeARivers committed:
перенесено завершение работы браузера в фикстуру

вынесены все урлы в константы и переиспользованы

первые локаторы изменены на более надежные

слипы и неявные ожидания заменены на expected_conditions

email для регистрации генерируется случайно

irisqul commented on tests/test_authorization_main.py:
```diff
@@ -0,0 +1,17 @@
+        time.sleep(1)
+        current_url = driver.current_url
+        assert current_url == 'https://stellarburgers.nomoreparties.site/'
+        driver.quit()
```
Нужно исправить: перенести завершение работы браузера в фикстуру

LikeARivers replied:
def test_authorization_main(self, driver):
        button_authorization_main = driver.find_element(*SBurgersLocators.BUTTON_AUTHORIZATION_MAIN).click()
        authorization_input_email = driver.find_element(*SBurgersLocators.AUTHORIZATION_INPUT_EMAIL).send_keys(*SBurgersTestData.AUTHARIZATION_EMAIL)
        authorization_input_password = driver.find_element(*SBurgersLocators.AUTHORIZATION_INPUT_PASSWORD).send_keys(*SBurgersTestData.REGISTRATION_PASSWORD_6_NUMBERS)
        authorization_login_button = driver.find_element(*SBurgersLocators.AUTHORIZATION_LOGIN_BUTTON).click()
        WebDriverWait(driver, 3).until(expected_conditions.presence_of_element_located((SBurgersLocators.MAKE_BURGER_TEXT)))
        current_url = driver.current_url
        assert current_url == URL

LikeARivers replied:
URL = "https://stellarburgers.nomoreparties.site/"
URL_LOGIN = "https://stellarburgers.nomoreparties.site/login"

irisqul commented on tests/test_authorization_main.py:
```diff
@@ -0,0 +1,17 @@
+        authorization_login_button = driver.find_element(*SBurgersLocators.AUTHORIZATION_LOGIN_BUTTON).click()
+        time.sleep(1)
+        current_url = driver.current_url
+        assert current_url == 'https://stellarburgers.nomoreparties.site/'
```
Нужно исправить: вынести все урлы в константы и переиспользовать их

LikeARivers replied:
URL = "https://stellarburgers.nomoreparties.site/"
URL_LOGIN = "https://stellarburgers.nomoreparties.site/login"

irisqul commented on locators.py:
```diff
@@ -0,0 +1,24 @@
+class SBurgersLocators:
+    PERSONAL_ACCOUNT_BUTTON_MAIN = (By.XPATH, ".//nav/a/p[text()='Личный Кабинет']") #кнопка Личный кабинет на главной странице
+    BUTTON_AUTHORIZATION_MAIN = (By.XPATH, "//button[text() = 'Войти в аккаунт']")  # кнопка Войти в аккаунт на главной странице
+    REGISTRATION_BUTTON = (By.XPATH, ".//main/div/div/p/a[text()='Зарегистрироваться']") #кнопка Зарегистрироваться на странице авторизации
```
Нужно исправить: ненадежный локатор. Пожалуйста найди более надежный путь до элементов, идеально если будет указано что-то специальное - вроде текста на кнопке или заголовка или иногда какие то атрибуты содержат отсылку к тому что за элемент. Не стоит ходить от root при описании локатора, поиск элементов не должен зависеть от верстки всей страницы

LikeARivers replied:
class SBurgersLocators:
    PERSONAL_ACCOUNT_BUTTON_MAIN = (By.XPATH, "//p[text()='Личный Кабинет']") #кнопка Личный кабинет на главной странице
    BUTTON_AUTHORIZATION_MAIN = (By.XPATH, "//button[text() = 'Войти в аккаунт']")  # кнопка Войти в аккаунт на главной странице
    REGISTRATION_BUTTON = (By.XPATH, "//a[text()='Зарегистрироваться']") #кнопка Зарегистрироваться на странице авторизации

irisqul commented on tests/test_authorization_main.py:
```diff
@@ -0,0 +1,17 @@
+        authorization_login_button = driver.find_element(*SBurgersLocators.AUTHORIZATION_LOGIN_BUTTON).click()
+        time.sleep(1)
+        current_url = driver.current_url
+        assert current_url == 'https://stellarburgers.nomoreparties.site/'
```
Нужно исправить: тест падает, хотя не должен, проверь пожалуйста ассерт. Также падают и другие подобные тесты, с сравнением урлов после авторизации, пожалуйста поправь их все.

LikeARivers replied:
def test_authorization_main(self, driver):
        button_authorization_main = driver.find_element(*SBurgersLocators.BUTTON_AUTHORIZATION_MAIN).click()
        authorization_input_email = driver.find_element(*SBurgersLocators.AUTHORIZATION_INPUT_EMAIL).send_keys(*SBurgersTestData.AUTHARIZATION_EMAIL)
        authorization_input_password = driver.find_element(*SBurgersLocators.AUTHORIZATION_INPUT_PASSWORD).send_keys(*SBurgersTestData.REGISTRATION_PASSWORD_6_NUMBERS)
        authorization_login_button = driver.find_element(*SBurgersLocators.AUTHORIZATION_LOGIN_BUTTON).click()
        WebDriverWait(driver, 3).until(expected_conditions.presence_of_element_located((SBurgersLocators.MAKE_BURGER_TEXT)))
        current_url = driver.current_url
        assert current_url == URL

irisqul commented on tests/test_forgot_password_authorization.py:
```diff
@@ -0,0 +1,18 @@
+
+class TestSBurgersAuthorizationForgotPassword:
+
+    def test_authorization_forgot_password(self, driver):
```
Нужно исправить: тест не запустился Unable to locate element: {"method":"xpath","selector":"//button[text()='Выход']"}

LikeARivers replied:
def test_authorization_forgot_password(self, driver):
        personal_area_button = driver.find_element(*SBurgersLocators.PERSONAL_ACCOUNT_BUTTON_MAIN).click()
        restore_password_button = driver.find_element(*SBurgersLocators.RESTORE_PASSWORD_BUTTON).click()
        authorization_button_restore_password = driver.find_element(*SBurgersLocators.AUTHORIZATION_BUTTON_RESTORE_PASSWORD).click()
        authorization_input_email = driver.find_element(*SBurgersLocators.AUTHORIZATION_INPUT_EMAIL).send_keys(*SBurgersTestData.AUTHARIZATION_EMAIL)
        authorization_input_password = driver.find_element(*SBurgersLocators.AUTHORIZATION_INPUT_PASSWORD).send_keys(*SBurgersTestData.REGISTRATION_PASSWORD_6_NUMBERS)
        authorization_login_button = driver.find_element(*SBurgersLocators.AUTHORIZATION_LOGIN_BUTTON).click()
        WebDriverWait(driver, 3).until(expected_conditions.presence_of_element_located((SBurgersLocators.MAKE_BURGER_TEXT)))
        current_url = driver.current_url
        assert current_url == URL

irisqul commented on tests/test_forgot_password_authorization.py:
```diff
@@ -0,0 +1,18 @@
+        authorization_input_email = driver.find_element(*SBurgersLocators.AUTHORIZATION_INPUT_EMAIL).send_keys(*SBurgersTestData.REGISTRATION_EMAIL)
+        authorization_input_password = driver.find_element(*SBurgersLocators.AUTHORIZATION_INPUT_PASSWORD).send_keys(*SBurgersTestData.REGISTRATION_PASSWORD_6_NUMBERS)
+        authorization_login_button = driver.find_element(*SBurgersLocators.AUTHORIZATION_LOGIN_BUTTON).click()
+        time.sleep(1)
```
Нужно исправить: здесь и далее: слипов и неявных ожиданий быть не должно. Необходимо использовать expected_conditions

LikeARivers replied:
def test_authorization_forgot_password(self, driver):
        personal_area_button = driver.find_element(*SBurgersLocators.PERSONAL_ACCOUNT_BUTTON_MAIN).click()
        restore_password_button = driver.find_element(*SBurgersLocators.RESTORE_PASSWORD_BUTTON).click()
        authorization_button_restore_password = driver.find_element(*SBurgersLocators.AUTHORIZATION_BUTTON_RESTORE_PASSWORD).click()
        authorization_input_email = driver.find_element(*SBurgersLocators.AUTHORIZATION_INPUT_EMAIL).send_keys(*SBurgersTestData.AUTHARIZATION_EMAIL)
        authorization_input_password = driver.find_element(*SBurgersLocators.AUTHORIZATION_INPUT_PASSWORD).send_keys(*SBurgersTestData.REGISTRATION_PASSWORD_6_NUMBERS)
        authorization_login_button = driver.find_element(*SBurgersLocators.AUTHORIZATION_LOGIN_BUTTON).click()
        WebDriverWait(driver, 3).until(expected_conditions.presence_of_element_located((SBurgersLocators.MAKE_BURGER_TEXT)))
        current_url = driver.current_url
        assert current_url == URL

irisqul commented on tests/test_go_to_sections.py:
```diff
@@ -0,0 +1,20 @@
+    def test_go_to_sections(self, driver):
+        stuffing_button = driver.find_element(*SBurgersLocators.STUFFING_BUTTON).click()
+        time.sleep(1)
+        assert driver.find_element(By.XPATH, "//h2[text() = 'Начинки']").is_displayed()
```
Нужно исправить: локаторы размещены в отдельном файле, нужно всегда их брать оттуда.

LikeARivers replied:
def test_go_to_sections_stuffing(self, driver):
        stuffing_button = driver.find_element(*SBurgersLocators.STUFFING_BUTTON).click()
        stuffing_button_activ = driver.find_element(*SBurgersLocators.STUFFING_BUTTON_ACTIV)
        assert 'current' in stuffing_button_activ.get_attribute('class')

irisqul commented on tests/test_go_to_sections.py:
```diff
@@ -0,0 +1,20 @@
+        assert driver.find_element(By.XPATH, "//h2[text() = 'Начинки']").is_displayed()
+        sauces_button = driver.find_element(*SBurgersLocators.SAUCES_BUTTON).click()
+        time.sleep(1)
+        assert driver.find_element(By.XPATH, "//h2[text() = 'Соусы']").is_displayed()
```
Нужно исправить: тест неатомарен, в одном тесте должна проверяться только одна функциональность и в среднем только один assert, после ассерта шагов уже быть не должно

LikeARivers replied:
def test_go_to_sections_stuffing(self, driver):
        stuffing_button = driver.find_element(*SBurgersLocators.STUFFING_BUTTON).click()
        stuffing_button_activ = driver.find_element(*SBurgersLocators.STUFFING_BUTTON_ACTIV)
        assert 'current' in stuffing_button_activ.get_attribute('class')

LikeARivers replied:
```
class TestSBurgersGoToSectionStuffing:
    def test_go_to_sections_stuffing(self, driver):
        stuffing_button = driver.find_element(*SBurgersLocators.STUFFING_BUTTON).click()
        stuffing_button_activ = driver.find_element(*SBurgersLocators.STUFFING_BUTTON_ACTIV)
        assert 'current' in stuffing_button_activ.get_attribute('class')


class TestSBurgersGoToSectionSauces:
    def test_go_to_sections_sauces(self, driver):
        sauces_button = driver.find_element(*SBurgersLocators.SAUCES_BUTTON).click()
        sauces_button_activ = driver.find_element(*SBurgersLocators.SAUCES_BUTTON_ACTIV)
        assert 'current' in sauces_button_activ.get_attribute('class')


class TestSBurgersGoToSectionBuns:
    def test_go_to_sections_buns(self, driver):
        sauces_button = driver.find_element(*SBurgersLocators.SAUCES_BUTTON).click()
        buns_button = driver.find_element(*SBurgersLocators.BUNS_BUTTON).click()
        buns_button_activ = driver.find_element(*SBurgersLocators.BUNS_BUTTON_ACTIV)
        assert 'current' in buns_button_activ.get_attribute('class')``
```

irisqul commented on tests/test_go_to_sections.py:
```diff
@@ -0,0 +1,20 @@
+    def test_go_to_sections(self, driver):
+        stuffing_button = driver.find_element(*SBurgersLocators.STUFFING_BUTTON).click()
+        time.sleep(1)
+        assert driver.find_element(By.XPATH, "//h2[text() = 'Начинки']").is_displayed()
```
Нужно исправить: наличие текста у этих элементов не говорит о том, что переход на таб произошел. У элемента при клике добавляется новый класс. Следует проверять его

LikeARivers replied:
`    STUFFING_BUTTON = (By.XPATH, "//span[text() = 'Начинки']") #кнопка Начинки в конструкторе

    SAUCES_BUTTON = (By.XPATH, "//span[text() = 'Соусы']") #кнопка Соусы в конструкторе
    
    BUNS_BUTTON = (By.XPATH, "//span[text() = 'Булки']") #кнопка Булки в конструкторе
    
    STUFFING_BUTTON_ACTIV = (By.XPATH, "//span[text() = 'Начинки']/parent::div")  #активация кнопки Начинки в конструкторе
    
    SAUCES_BUTTON_ACTIV = (By.XPATH, "//span[text() = 'Соусы']/parent::div")  #активация кнопки Соусы в конструкторе
    
    BUNS_BUTTON_ACTIV = (By.XPATH, "//span[text() = 'Булки']/parent::div")  #активация кнопки Булки в конструкторе
    
    MAKE_BURGER_TEXT = (By.XPATH, "//h1[text()='Соберите бургер']") #текст Соберите бургер на главной странице`

irisqul commented on tests/test_successful_registration.py:
```diff
@@ -0,0 +1,20 @@
+        personal_area_button = driver.find_element(*SBurgersLocators.PERSONAL_ACCOUNT_BUTTON_MAIN).click()
+        registration_button = driver.find_element(*SBurgersLocators.REGISTRATION_BUTTON).click()
+        registration_input_name = driver.find_element(*SBurgersLocators.REGISTRATION_INPUT_NAME).send_keys(*SBurgersTestData.REGISTRATION_NAME)
+        registration_input_email = driver.find_element(*SBurgersLocators.REGISTRATION_INPUT_EMAIL).send_keys(*SBurgersTestData.REGISTRATION_EMAIL)
```
Нужно исправить: одноразовый тест. Чтобы проходила регистрация нового пользователя email каждый тест должен быть новый

LikeARivers replied:
```
    def test_registration(self, driver):
        personal_area_button = driver.find_element(*SBurgersLocators.PERSONAL_ACCOUNT_BUTTON_MAIN).click()
        registration_button = driver.find_element(*SBurgersLocators.REGISTRATION_BUTTON).click()
        registration_input_name = driver.find_element(*SBurgersLocators.REGISTRATION_INPUT_NAME).send_keys(*SBurgersTestData.REGISTRATION_NAME)
        registration_input_email = driver.find_element(*SBurgersLocators.REGISTRATION_INPUT_EMAIL).send_keys(*SBurgersTestData.REGISTRATION_EMAIL)
        registration_input_password = driver.find_element(*SBurgersLocators.REGISTRATION_INPUT_PASSWORD).send_keys(*SBurgersTestData.REGISTRATION_PASSWORD_6_NUMBERS)
        registration_finish_button = driver.find_element(*SBurgersLocators.REGISTRATION_FINISH_BUTTON).click()
        WebDriverWait(driver, 6).until(expected_conditions.url_to_be("https://stellarburgers.nomoreparties.site/login"))
        current_url = driver.current_url
        assert current_url == URL_LOGIN
```

LikeARivers committed:
перенесено завершение работы браузера в фикстуру

вынесены все урлы в константы и переиспользованы

первые локаторы изменены на более надежные

слипы и неявные ожидания заменены на expected_conditions

исправлен тест test_go_to_sections.py

LikeARivers committed:
Merge remote-tracking branch 'origin/develop' into develop

# Conflicts:
#	conftest.py
#	data.py
#	locators.py
#	settings.py
#	tests/test_authorization_main.py
#	tests/test_authorization_personal_account.py
#	tests/test_authorization_registration_form.py
#	tests/test_exit.py
#	tests/test_forgot_password_authorization.py
#	tests/test_go_to_sections.py
#	tests/test_successful_registration.py
#	tests/test_transition_constructor.py
#	tests/test_transition_logo.py
#	tests/test_transition_personal_account.py
#	tests/test_unsuccessful_registration.py

LikeARivers committed:
Update README.md
